protocol/tx/entry.go

```diff
@@ -73,6 +97,10 @@ func writeForHash(w io.Writer, c interface{}) error {
 	case bc.AssetID:
 		_, err := w.Write(v[:])
 		return errors.Wrap(err, "writing bc.AssetID for hash")
+
+	case *EntryRef:
```

Comment:
(This isn't a concrete proposal, just an idea that I am wondering if we should consider.)